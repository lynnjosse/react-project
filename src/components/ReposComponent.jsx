import React from 'react'
import PropTypes from 'prop-types'
import Immutable from 'immutable'
import ReposRecord from './../reducers/Repos'

const ReposComponent = ({repos, retryFetchingRepos}) => (
  <div>
    {repos.status === 'FETCHING' ? (
      <div className = "alert alert-info">
        <span className = "velocity-spinner-small-inline" />
        <span>...getting your repos</span>
      </div   >
    ) :

      repos.status ==='ERROR' ? (
        <div className = "alert alert-danger">

            something went wrong while getting your damn repos
            <button
              type= "button"
              className = "btn btn-danger pull-right"
              onClick={retryFetchingRepos}
            >
              Retry
            </button>

        </div>
        ) : repos.status ==='SUCCESS' ? (
          <ul>
            {repos.value.map(repo => (
              <a key = {repo.id} href = {repo.html_url} className = "list-group-item">
                {repo.name}
              </a>

            ))}
          </ul>
        ) : null
    }
  </div>
  );

ReposComponent.displayName = 'ReposComponent';

ReposComponent.propTypes = {
  repos: PropTypes.instanceOf(Immutable.Record).isRequired,
  retryFetchingRepos: PropTypes.func.isRequired
}
export default ReposComponent
