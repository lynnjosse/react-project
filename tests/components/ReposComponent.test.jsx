//we need to be able to render the component we care about (only)
//we need to be able to convert it into a json format)
//enzyme takes care of the shallow rendering
//npm install --save-dev enzyme enzyme-adapter-react-16 enzyme-to-json
//import shallow makes it only render the component we are testing, not the ones within it
//shallow returns a wrapper
//jest.fn is a mock function - we don't have to actually supply a function here

import ReposComponent from './../../src/components/ReposComponent'
import {shallow} from 'enzyme'
import React from 'react'
import ReposRecord from './../../src/reducers/Repos'

jest.unmock('./../../src/components/ReposComponent')
jest.unmock('./../../src/reducers/Repos')


describe('ReposComponent', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(
      <ReposComponent repos={new ReposRecord()} retryFetchingRepos={jest.fn()} />
    )
  })

  it('default view is empty', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('displays alert when repos are fetching',() => {
    const newReposRecord = new ReposRecord({
      status: 'FETCHING'
    })
    wrapper.setProps({repos: newReposRecord})
    expect(wrapper).toMatchSnapshot()
  })


})
