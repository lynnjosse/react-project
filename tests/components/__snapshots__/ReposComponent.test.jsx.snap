// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReposComponent default view is empty 1`] = `undefined`;
